build/configuration-properties-collector/src/main/java/org/hibernate/search/configuration/properties/collector/impl/AsciiDocWriter.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.configuration.properties.collector.impl;
+
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class AsciiDocWriter implements BiConsumer<Map<String, ConfigurationProperty>, Writer> {
+
+	private final Optional<String> moduleName;
+	private final Predicate<Map.Entry<String, ConfigurationProperty>> filter;
+
+	public AsciiDocWriter(String moduleName, Predicate<Map.Entry<String, ConfigurationProperty>> filter) {
+		this.moduleName = Optional.ofNullable( moduleName );
+		this.filter = filter;
+	}
+
+	@Override
+	public void accept(Map<String, ConfigurationProperty> propertyMap, Writer writer) {
+		List<Map.Entry<String, ConfigurationProperty>> entries = propertyMap.entrySet().stream()
+				.filter( filter )
+				.collect( Collectors.toList() );
+		if ( entries.isEmpty() ) {
+			// nothing to write - return fast.
+			return;
+		}
+		try {
+			moduleName.ifPresent( name -> tryToWriteLine( "== ", name, writer ) );
+			writer.write( '\n' );
+			for ( Map.Entry<String, ConfigurationProperty> entry : entries ) {
+				Iterator<String> keys = entry.getValue().key().resolvedKeys().iterator();
+				writer.write( '`' );
+				writer.write( keys.next() );
+				writer.write( '`' );
+				writer.write( "::\n" );
+
+				boolean hasMultipleKeys = false;
+				if ( keys.hasNext() ) {
+					hasMultipleKeys = true;
+					writer.write( "Other variants: " );
+				}
+				while ( keys.hasNext() ) {
+					writer.write( '`' );
+					writer.write( keys.next() );
+					writer.write( '`' );
+					if ( keys.hasNext() ) {
+						writer.write( ", " );
+					}
+				}
+
+				if ( hasMultipleKeys ) {
+					writer.write( "\n+\n" );
+				}
+
+				// using inline passthrough for javadocs to not render HTML.
+				writer.write( "+++ " );
+				writer.write( entry.getValue().javadoc() );
+				writer.write( " +++ " );
+
+				String defaultValue = Objects.toString( entry.getValue().defaultValue(), "" );
+				if ( !defaultValue.isBlank() ) {
+					writer.write( "\n+\n" );
+					writer.write( "Default value: `" );
+					writer.write( defaultValue );
+					writer.write( '`' );
+				}
+
+				writer.write( '\n' );
```

Comment:
I'll give this [collapsible section](https://docs.asciidoctor.org/asciidoc/latest/blocks/collapsible/ ) a try maybe it'll be ok...
something along the lines:
```
.Other configuration variants (Click here):
[%collapsible]
====
`a.b.c`, `d.f.g` ... 
====
```

Reply:
it looks like this:
![image](https://user-images.githubusercontent.com/4004823/213148385-4c9d055a-ce4b-4ba3-8500-c93e2eff91f9.png)
Collapsed and expended variants. Seems ok, except for the label 😄 - the "click here" part is so-so, but I'm not sure if we remove it that it would be obvious for the users to click it (even though it has a hand cursor on hover)

Reply:
LGTM, but I'd use `Variants of this configuration property (Click here):`. YMMV :)